src/extensions/otel_viewer_extension/OtelData.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using Newtonsoft.Json;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+using Newtonsoft.Json.Linq;
+using Google.Protobuf;
+using Azure.AI.Details.Common.CLI;
+
+namespace Azure.AI.Details.Common.CLI.Extensions.Otel
+{
+    public class OtelData
+    {
+        public static async Task<string> GetTrace(string requestId)
+        {
+            // Endpoint URL with the request ID
+            string url = $"https://spanretriever1-webapp.whitehill-dc9ec001.westus.azurecontainerapps.io/getspan/{requestId}";
```

Comment:
eventually, how will this get retrieved? for the actual endpoint for the user to use?

Reply:
the web service will be deployed as an Azure Container App so the endpoint will be public and authentication will happen within the web service to determine if the user can access the spans